wallet/src/main/scala/org/bitcoins/wallet/internal/FundTransactionHandling.scala

```diff
@@ -82,7 +82,7 @@ trait FundTransactionHandling extends WalletLogger { self: LockedWalletApi =>
 
     val txBuilderF = for {
       addrInfosWithUtxo <- addrInfosWithUtxoF
-      change <- changeAddrF
+      change <- getNewChangeAddress(fromAccount)
```

Comment:
This is import that it is moved here, because if we fail earlier in coin selection the background thread that generates address will continue to attempt to generate an address. This means we get a bunch of exception stack traces in unit tests when we _want_ to fail funding a transaction, but we continue attempting to generate an address. When the address is inserted into the database, we have already torn down the wallet database which causes an exception.